app/src/main/java/com/therap/amin/currencyconverter/MainActivity.java

```diff
@@ -1,241 +1,174 @@
-                        } else if (fileProcessor.calculateConversionRate(from, to) != -1) {
-                            conversionVal = fileProcessor.calculateConversionRate(from, to);
-                        }
-                        double convertedAmount = Double.parseDouble(s.toString());
-                        convertedAmount *= conversionVal;
-                        tvOutputCurrencyValue.setText(String.format("%.2f", convertedAmount));
-                    } else {
-                        tvOutputCurrencyValue.setText("0");
-                    }
-                }
-
-                @Override
-                public void afterTextChanged(Editable s) {
-
-                }
-            });
         }
-
     }
 
     @Override
     public boolean onCreateOptionsMenu(Menu menu) {
+        this.menu = menu;
+        MenuInflater menuInflater = getMenuInflater();
+        menuInflater.inflate(R.menu.main, menu);
         if (tabletSize) {
-            return false;
+            MenuItem item = menu.findItem(R.id.set_currency_rates);
+            item.setVisible(false);
+            this.invalidateOptionsMenu();
         } else {
-            MenuInflater menuInflater = getMenuInflater();
-            menuInflater.inflate(R.menu.main, menu);
-            return true;
+            RoboFragment fragment = (RoboFragment) getSupportFragmentManager().findFragmentById(R.id.fragmentcontainer);
+            setMenus(fragment);
+            this.invalidateOptionsMenu();
         }
+        return true;
     }
 
     @Override
     public boolean onOptionsItemSelected(MenuItem item) {
-        if (tabletSize) {
-            return super.onOptionsItemSelected(item);
-        } else {
-            switch (item.getItemId()) {
-                case R.id.menu_currencyvalues:
-                    Intent intent = new Intent(MainActivity.this, CurrencyValueSaverActivity.class);
-                    String from = fromCurrencySpinner.getSelectedItem().toString();
-                    String to = toCurrencySpinner.getSelectedItem().toString();
-                    intent.putExtra(Constants.FROM_CURRENCY_KEY, from);
-                    intent.putExtra(Constants.TO_CURRENCY_KEY, to);
-                    startActivityForResult(intent, 1);
-                    return true;
-
-                default:
-                    return super.onOptionsItemSelected(item);
-            }
+
+        switch (item.getItemId()) {
+            case R.id.set_currency_rates:
+                replaceFragment(new CurrencyValueSaverFragment());
+                setMenus(new CurrencyValueSaverFragment());
+                return true;
+
+            case R.id.load_currency_rates:
+                final ProgressDialog progressDialog = new ProgressDialog(MainActivity.this);
+                progressDialog.setMessage("Loading...");
+                progressDialog.show();
+                FetchCurrencyRates.get(Constants.URL, null, new JsonHttpResponseHandler() {
```

Comment:
Move the web service call part to a service named "CurrencyConverterService".
Pass a handler from this activity to that service. On success or failure call, that handler method would be invoked.